client/lib/store-sandbox-helper/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ToggleControl } from '@wordpress/components';
+import { useState, useEffect } from 'react';
+import ReactDom from 'react-dom';
+import useStoreSandboxStatusQuery from 'calypso/data/store-sandbox/use-store-sandbox-status';
+import wp from 'calypso/lib/wp';
+
+import './style.scss';
+
+interface StoreSandboxQueryResponse {
+	store_sandbox_cookie_name: string | undefined;
+	store_sandbox_expiration_time: number | undefined;
+	is_disabled: boolean | undefined;
+}
+
+export function StoreSandboxHelper() {
+	const { data: isSandboxed = false } = useStoreSandboxStatusQuery();
+	const [ isStoreSandboxed, setIsStoreSandboxed ] = useState( isSandboxed );
+
+	useEffect( () => {
+		setIsStoreSandboxed( isSandboxed );
+	}, [ isSandboxed ] );
+
+	const onToggleStoreSandbox = () => {
+		wp.req.post(
+			{
+				path: `/store-sandbox/${ isStoreSandboxed ? 'disable' : 'enable' }`,
+				apiNamespace: 'wpcom/v2',
+			},
+			{},
+			( error: string, data: StoreSandboxQueryResponse ) => {
+				if ( ! error && data ) {
+					setIsStoreSandboxed( !! data?.store_sandbox_cookie_name );
```

Comment:
Should we automatically refresh the page here?  In general, store sandbox doesn't really take effect without some kind of page refresh, so it kind of feels broken now because you use the toggle and nothing happens.

Reply:
Great suggestion - implemented in ddf9a96ca2e07e7bf142ae0024f45f3ef910b5dd